addons/account/tests/test_account_move_reconcile.py

```diff
@@ -4000,6 +4000,104 @@ def test_reconcile_cash_basis_tax_grid_multi_taxes(self):
             {'debit': 0.0,      'credit': 0.01,     'tax_ids': [],              'tax_tag_ids': self.tax_tags[5].ids,    'account_id': self.tax_account_2.id},
         ])
 
+    def test_matching_number_full_reconcile(self):
+        currency = self.currency_data['currency']
+        line_a = self.create_line_for_reconciliation(1000, 1000, currency, '2016-01-01')
+        line_b = self.create_line_for_reconciliation(-1000, -1000, currency, '2016-01-01')
+        (line_a + line_b).reconcile()
+        self.assertFullReconcile(line_a.full_reconcile_id, (line_a + line_b))
+        self.assertEqual(line_a.matching_number, str(line_a.full_reconcile_id.id))
+        self.assertEqual(line_a.matching_number, line_b.matching_number)
+
+    def test_matching_number_partial_single_reconcile(self):
+        currency = self.currency_data['currency']
```

Comment:
all test are written using a foreign currency but a rate of 1
Do you mean `self.env.company.currency_id` here instead?

Reply:
Doesn't really matter for the purpose of the test, does it?
But yes, the company's currency would probably be cleaner :+1: